src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length - 1; i++) {
```
В таком случае тебе не надо будет на каждом шаге делать проверку `if (i === this.length - 1) {`
Ты просто выполнишь присвоение последнего элемента вне цикла